sdk/tests/conformance2/rendering/multisampling-fragment-evaluation.html

```diff
@@ -0,0 +1,164 @@
+// "Polygon rasterization produces a fragment for each framebuffer pixel with one or more sample points that satisfy
+// the point sampling criteria described in section 3.6.1."
+
+debug("Regression test for <a href='http://crbug.com/682815'>http://crbug.com/682815</a>");
+
+function runTest(testParams) {
+    let canvas = document.createElement('canvas');
+    canvas.width = 1;
+    canvas.height = 1;
+    let gl = wtu.create3DContext(canvas, {antialias: false}, 2);
+
+    // Find the supported samples for a multisampled renderbuffer of the appropriate internal format.
+    let samples = gl.getInternalformatParameter(gl.RENDERBUFFER, gl[testParams.internalformat], gl.SAMPLES);
+    if (!samples || !samples.length) {
+        testFailed("Could not query supported sample counts for required multisampling format " + testParams.internalformat);
+        return;
+    }
+
+    // Note that supported sample counts are required to be reported in descending order.
+    debug('Testing with sample count ' + samples[0]);
+    // Create a framebuffer with a multisampled renderbuffer with the maximum supported number of samples.
+    let rb = gl.createRenderbuffer();
+    gl.bindRenderbuffer(gl.RENDERBUFFER, rb);
+    gl.renderbufferStorageMultisample(gl.RENDERBUFFER, samples[0], gl[testParams.internalformat], 1, 1);
+    let fb = gl.createFramebuffer();
+    gl.bindFramebuffer(gl.FRAMEBUFFER, fb);
+    gl.framebufferRenderbuffer(gl.FRAMEBUFFER, gl.COLOR_ATTACHMENT0, gl.RENDERBUFFER, rb);
+    if (gl.checkFramebufferStatus(gl.FRAMEBUFFER) != gl.FRAMEBUFFER_COMPLETE) {
+        testFailed("Rendering to a multisampled renderbuffer of format " + testParams.internalformat + " is required.");
+        return;
+    }
+
+    // Create a program that will choose between one of different possible colors in the fragment shader.
+    // It should be evaluated only once per framebuffer pixel, so only one of the colors will end up in the framebuffer.
+    // However, if the multisampling mode is incorrectly implemented by supersampling, the samples may have different
+    // colors.
+    let program = wtu.setupProgram(gl, ["vshader", "fshader"], ["aPosition"]);
+
+    // Render one triangle using the program. The triangle needs to extend far outside the viewport on all sides, so
+    // that we can safely assume all samples fall inside the triangle. GLES 3.0.5:
+    // "The sample points associated with a pixel may be located inside or outside of the unit square that is considered to bound the pixel."
+    // Here we assume that sample points are less than 9999 pixels away from the pixel they are associated with.
+    let buffer = gl.createBuffer();
+    gl.bindBuffer(gl.ARRAY_BUFFER, buffer);
+    gl.bufferData(gl.ARRAY_BUFFER, new Float32Array([ 10000,  30000,
+                                                     -30000, -10000,
+                                                      10000, -10000]), gl.STATIC_DRAW);
```

Comment:
In practice a couple of pixels is probably enough, but since this is testing an area of the spec that is fairly open to interpretation, I decided to make it conservative.